apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/backend/ir/BackendIr.kt

```diff
@@ -99,46 +97,38 @@ internal data class BackendIr(
       }
     }
 
-    data class Variable(val name: String, val inverted: Boolean) : Condition() {
-    }
+    data class Variable(val name: String, val inverted: Boolean) : Condition()
   }
 
-  sealed class Fragment {
-    abstract val name: String
-    abstract val fields: List<Field>
-    abstract val possibleTypes: Set<IntrospectionSchema.TypeRef>
-    abstract val selectionKeys: Set<SelectionKey>
-    abstract val description: String?
-
-    data class Interface(
-        override val name: String,
-        override val fields: List<Field>,
-        override val possibleTypes: Set<IntrospectionSchema.TypeRef>,
-        override val selectionKeys: Set<SelectionKey>,
-        override val description: String?,
-        val typeCondition: IntrospectionSchema.TypeRef,
-    ) : Fragment()
-
-    data class Implementation(
-        override val name: String,
-        override val fields: List<Field>,
-        override val possibleTypes: Set<IntrospectionSchema.TypeRef>,
-        override val selectionKeys: Set<SelectionKey>,
-        override val description: String?,
-    ) : Fragment()
+  data class Fragments(
+      val fragments: List<Fragment>,
+      val accessors: Map<String, SelectionKey>,
+  ) : List<Fragment> by fragments
+
+  data class Fragment(
+      val name: String,
```

Comment:
What about?

```suggestion
      val typeCondition: String,
```

As `name` could be mistaken with the fragment spread name

Reply:
Got it. It's not directly `typeCondition` but `typeCondition.capitalize` so it's the name of the generated class. Oh well, naming is hard... Maybe we could postpone the capitalization to the codegen phase and use only `typeCondition` in BackendIr?

Reply:
We could but there some logic around selection key comparison + formatting names for accessors, default implementation etc. Even if remove it there will be only 2 places at most.